include/kernel.h

```diff
@@ -3040,6 +3057,112 @@ static inline s32_t k_delayed_work_remaining_get(struct k_delayed_work *work)
 	return __ticks_to_ms(z_timeout_remaining(&work->timeout));
 }
 
+/**
+ * @brief Initialize a triggered work item.
+ *
+ * This routine initializes a workqueue triggered work item, prior to
+ * its first use.
+ *
+ * @param work Address of triggered work item.
+ * @param handler Function to invoke each time work item is processed.
+ *
+ * @return N/A
+ * @req K-TWORK-001
+ */
+extern void k_triggered_work_init(struct k_triggered_work *work,
+				  k_work_handler_t handler);
+
+/**
+ * @brief Submit a triggered work item.
+ *
+ * This routine schedules work item @a work to be processed by workqueue
+ * @a work_q when one of the given @a events is signaled. The routine
+ * initiates internal poller for the work item and then returns to the caller.
+ * Only when one of the watched events happen the work item is actually
+ * submitted to the workqueue and becomes pending.
+ *
+ * Submitting a previously submitted triggered work item that is still
+ * waiting for the event cancels the existing submission and reschedules it
+ * the using the new event list. Note that this behavior is inherently subject
+ * to race conditions with the pre-existig triggered work item and work queue,
+ * so care must be taken to synchronize such resubmissions externally.
+ *
+ * @note Can be called by ISRs.
+ *
+ * @warning
+ * Provided array of events as well as a triggered work item must not be
+ * modified until the item has been processed by the workqueue.
+ *
+ * @param work_q Address of workqueue.
+ * @param work Address of delayed work item.
+ * @param events An array of pointers to events which trigger the work.
+ * @param num_events The number of events in the array.
+ *
+ * @retval 0 Work item started watching for events.
+ * @retval -EINVAL Work item is being processed or has completed its work.
+ * @retval -EADDRINUSE Work item is pending on a different workqueue.
+ * @req K-TWORK-001
```

Comment:
remove `@req` annotations throughout.

Reply:
Done.

Reply:
I'm still seeing them, but it looks like there are a lot of pre-existing ones, so leaving the new ones in too doesn't seem worth fussing about.

Reply:
Grrrr. I not staged kernel.h to commit. Now it should be OK.